packages/block-library/src/navigation/edit.js

```diff
@@ -115,6 +113,13 @@ function Navigation( {
 	const [ isResponsiveMenuOpen, setResponsiveMenuVisibility ] = useState(
 		false
 	);
+	const hasSubmenuIndicatorSetting =
+		useSetting( 'hasSubmenuIndicatorSetting' ) ?? true; // retain original prop default of "true" if there is no setting defined.
```

Comment:
The benefit of this approach is that `useSetting` is much more declarative. If the a setting is add/removed/changed then it's much easier for the Nav Editor to declare support (or not!).

Also it's much harder for developers working on the Nav block to accidentally break the Nav Editor because making a change to a setting is a larger undertaking.